crates/libcontainer/src/rootless.rs

```diff
@@ -7,6 +7,42 @@ use std::path::Path;
 use std::process::Command;
 use std::{env, path::PathBuf};
 
+// Wrap the uid/gid path function into a struct for dependency injection. This
+// allows us to mock the id mapping logic in unit tests.
+#[derive(Debug, Clone, Default)]
+pub struct RootlessIDMapper {
+    pub path: Option<PathBuf>,
```

Comment:
Since I feel this name `path` is too abstract, it was difficult to determine what field it was. Can we name it something more specific? I like the mechanism itself 👍